### PR TITLE
refactor (ldap): add new config fields for include/excluding user attrs

### PR DESCRIPTION
**Pull request by jimlambrt**
IncludeUserAttributes optionally specifies that the authenticating user's attributes be included an authentication AuthResult.

ExcludedUserAttributes optionally defines a set of user attributes to be excluded when an authenticating user's attributes are included in an AuthResult (see: Config.IncludeUserAttributes or the WithUserAttributes() option).

Note: the default password attribute for both openLDAP (userPassword) and AD (unicodePwd) will always be excluded.

### CODE REVIEW

File: ldap/README.md [hunk 1]
```diff
@@ -102,10 +102,26 @@ information on UPN can be found
...
+distinguished name defined for `BindDN` is used for the group search. Otherwise,
+the authenticating user is used to perform the group search.
+
+### User Attributes
+Using configuration you can choose to optionally include an authenticated
+user's DN and entry attributes in the results of an authentication request.  
+
+* `IncludeUserAttributes` (bool, optional) - If true, specifies that the
+  authenticating user's DN and attributes be included an authentication
+  AuthResult. Note: the default  password attribute for both openLDAP
```
Comment (austingebauer) [inline ldap/README.md:121]:
```suggestion
  AuthResult. Note: the default password attribute for both openLDAP
```

File: ldap/client.go [hunk 2]
```diff
@@ -307,10 +329,19 @@ func (c *Client) getUserAttributes(userDN string) ([]Attribute, error) {
...
 	for _, a := range userEntry.Attributes {
-		attributes = append(attributes, Attribute{
-			Name: a.Name,
-			Vals: a.Values,
-		})
+		switch {
+		// exclude the default openLDAP password attribute
+		case strings.EqualFold(a.Name, DefaultOpenLDAPUserPasswordAttribute):
+		// exclude the default AD password attribute
+		case strings.EqualFold(a.Name, DefaultADUserPasswordAttribute):
```
Comment (austingebauer) [inline ldap/client.go:336]: 👍

File: ldap/config.go [hunk 3]
```diff
@@ -29,6 +29,14 @@ const (
 
 	// DefaultTLSMaxVersion for the ClientConfig.TLSMaxVersion
 	DefaultTLSMaxVersion = "tls12"
+
+	// DefaultOpenLDAPUserPasswordAttribute defines the attribute name for the
+	// openLDAP default password attribute which will always be excluded f
```
Comment (austingebauer) [inline ldap/config.go:34]: Additional "f" here:

```suggestion
	// openLDAP default password attribute which will always be excluded
```

File: ldap/config.go [hunk 4]
```diff
@@ -29,6 +29,14 @@ const (
...
+	DefaultOpenLDAPUserPasswordAttribute = "userPassword"
+
+	// DefaultADUserPasswordAttribute defines the attribute name for the
+	// AD default password attribute
```
Comment (austingebauer) [inline ldap/config.go:38]: Might want to add "which will always be excluded" similar to above?

**Review by austingebauer — APPROVED** [review]
A few small comments. Otherwise, LGTM 👍

File: ldap/config.go (hunk 3 above)
Comment (jimlambrt) [reply ldap/config.go:34]: ty!

File: ldap/config.go (hunk 4 above)
Comment (jimlambrt) [reply ldap/config.go:38]: ty!

File: ldap/README.md (hunk 1 above)
Comment (jimlambrt) [reply ldap/README.md:121]: ty!

**Pull request merged**